rustls-mbedpki-provider/src/lib.rs

```diff
@@ -31,12 +31,21 @@
 // Enable documentation for all features on docs.rs
 #![cfg_attr(docsrs, feature(doc_cfg, doc_auto_cfg))]
 #![cfg_attr(bench, feature(test))]
+#![cfg_attr(not(test), no_std)]
```

Comment:
This already takes effect on current code as you can see, I made some changes to other files to avoid using `std`.

Do you mean you have some code depending on `std` need to be upstream in the future?

Reply:
Yeah, I mean our dependencies require std anyway right now. I'm not objecting to this, just want to understand the motivation for it.

Reply:
Oh, I see. Current dependencies still depend on `std`.
It makes sense. 
Let just keep the changes in this PR.

There will be some more work to fully support `no_std`, such as removing `std` feature flag on `mbedtls`.

Reply:
Ok. Sounds good!